RFC/src/RFC-0140 Syncing and seeding.md

```diff
@@ -0,0 +1,131 @@
+community. The release of this document is intended solely for review and discussion by the community regarding the
+technological merits of the potential system outlined herein.
+
+## Goals
+
+This document describes the process of Syncing, seeding, pruning and cut-through.
+
+## Related RFCs
+
+* [RFC-0110: Base Nodes](RFC-0110_BaseNodes.md)
+
+## Descriptions
+
+### Syncing
+
+When a new node comes online, looses connection or encounters a fork that is longer than is longer than it can solve, it must enter syncing mode. This will allow it to recover its state to the newest up to date state. Sycning can be diveded into 2 [SynchronisationStrategy]s, complete sync and sync. Complete sync will involve that the node communicates with an archive node to get the complete history of every single block from genesis block. Sync will involve the node getting every block from its [puning horizon](pruninghorizon) to current head, as well as every block header from genesis block. 
+
+#### Complete Sync
+
+Complete sync is only available from archive nodes, as these will be the only nodes that will be able to supply the complet history required to sync every block with every transaction from genesis block up onto [current head](currenthead). 
+
+
+
+#### Syncing process
+
+The syncing process is done in the following steps:
+
+1. Set its [SynchronisationState] to `Synchronising`.
+2. Load a bootstrap list of peers from a configuration file, or a cached list.
+3. For each peer in this list:
+   1. Establish a connection with the peer.
+   2. Request a peer list from that peer.
+   3. Request information about the most recent chain state (total accumulated work, block height, etc.) from the peer.
+4. Choose the longest chain based on pow. 
+5. Download all headers from genesis block up onto [current head](currenthead), an validate the headers as you receive them.
+6. Download all blocks from  [pruning horizon](pruninghorizon) up onto [current head](currenthead), if the node is doing a complete sync, the pruninghorizon will just be infinite, which means you will download all blocks ever created.
+7. Validate blocks as if they where just mined and then received, in order of ascending age. 
+
+After this process the node will be in sync and able to process blocks and transaction normally. 
+
+#### Keeping in sync
+
+The node should periodically test its peers with ping messages to ensure that they are alive. When a node sends a ping message, it should enclude the current total pow, hash of the [current head](currenthead) and genesis block hash of its own current longest chain in the ping message. The receiving node will replay with a pong message also including the total pow, [current head](currenthead) and genesis block hash of its longest chain. If the two chains dont match up, the node with the lowest pow has the responsiblity to ask the peer for syncing information. 
```

Comment:
Do I read this correctly? 
> genesis block hash of its own current longest chain

If I have a fork at block > 0, I'll always pass the Genesis block hash (i.e. block 0)?

Reply:
:+1: